### PR TITLE
Simplifications and fixes to multicore systhreads implementation (0.5/3)

### PR DESCRIPTION
**Pull request by gadmm**
This contains commits from #11271 except the one that fails the test suite on Alpine.

### CODE REVIEW

**Comment by gasche** [comment]
Precheck running at https://ci.inria.fr/ocaml/job/precheck/729/

**Comment by xavierleroy** [comment]
Precheck is OK.  One unrelated failure with Omnios (will check separately) but Alpine, PPC32 and PPC64 are back.

File: runtime/caml/domain.h [hunk 1]
```diff
@@ -70,6 +70,7 @@ void caml_reset_young_limit(caml_domain_state *);
 
 CAMLextern void caml_reset_domain_lock(void);
 CAMLextern int caml_bt_is_in_blocking_section(void);
+CAMLexport int caml_bt_is_self(void);
```
Comment (dra27) [inline runtime/caml/domain.h]: This fixes some of the Windows failures, but not all:
```suggestion
CAMLextern int caml_bt_is_self(void);
```

Comment (dra27) [reply runtime/caml/domain.h]: Ah, to expand - the Windows workers in Jenkins are running binutils 2.34, so don't surface the bug which this fixes. The other failure I'm seeing locally is in the `tests/output-complete-obj/main.ml`, but that doesn't appear to be this PR

Comment (dra27) [reply runtime/caml/domain.h]: (indeed - the failure with `tests/output-complete-obj/main.ml` seems to be caused by binutils 2.38, which I updated to two days ago <sigh>)

Comment (gadmm) [reply runtime/caml/domain.h]: Thanks for catching this. This is fixed now.

**Comment by gadmm** [comment]
I have arranged that no further changes are needed with this PR, so I propose to go ahead.

**Review by gasche — APPROVED** [review]
I'll approve based on:
- KC's and Engil's review of the previous PR #11271 
- the fact thaI the INRIA CI is now happy

This will be trunk-only for now: let's wait to see where we are -- and to understand the cause of the bug -- before pushing things in 5.0.

**Comment by gasche** [comment]
Backported to 5.0 as 42bf679e5e03ad9841d1ccba305a74f266fd98d6.

**Pull request merged**